base/strings/util.jl

```diff
@@ -464,11 +464,103 @@ function hex2bytes(s::AbstractString)
 end
 
 """
+    hex2bytes(s::Vector{UInt8})
+
+Convert the hexadecimal bytes array to its binary representation. Returns an
+`Array{UInt8,1}`, i.e. an array of bytes.
+"""
+@inline function hex2bytes(s::Vector{UInt8})
+    d = Vector{UInt8}(div(length(s), 2))
```

Comment:
Not needed as the underlying method will throw exception. Per @StefanKarpinski  all the interfaces are changed to AbstractVector. Many of comments may not be relevant in that case.